### PR TITLE
handling new seasons

### PR DESCRIPTION
**Pull request by tivvit**

### CODE REVIEW

**Comment by malytomas** [comment]
will this behave correctly when the season is changed directly in database (while the server was not running)?

**Comment by tivvit** [comment]
> will this behave correctly when the season is changed directly in the database (while the server was not running)?

Good question. It depends on your definition of correctly.

Not running game server has no state. Therefore it loads the game state from the DB and does the necessary init when started.  

I do belive that there is no edge-case but maybe you had something specific in mind?

File: server/stycoon/main_loop.go [hunk 1]
```diff
@@ -23,8 +23,23 @@ func (game *Game) callUpdate() {
...
+	}
+	err = game.Init()
+	if err != nil {
+		log.Error().Err(err).Msg("Init game after new season failed")
+	}
+}
+
 func (game *Game) MainLoop(ctx context.Context, wg *sync.WaitGroup) {
 	ticker := time.NewTicker(TickDuration)
+	seasonDuration := 2 * int64(time.Hour.Seconds())
```
Comment (evamayerova) [inline server/stycoon/main_loop.go:42]: add a TODO to make this configurable

**Review by evamayerova — APPROVED** [review]

File: server/stycoon/main_loop.go (hunk 1 above)
Comment (tivvit) [reply server/stycoon/main_loop.go:42]: done

**Pull request merged**